core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -265,7 +265,7 @@ private[spark] object HighlyCompressedMapStatus {
     val threshold =
       if (accurateBlockSkewedFactor > 0) {
         val sortedSizes = uncompressedSizes.sorted
-        val medianSize: Long = Utils.median(sortedSizes)
+        val medianSize: Long = Utils.median(sortedSizes, true)
```

Comment:
only in this call site `alreadySorted` is `true`, what about only changing this place to utilize the order?

Reply:
Hi, @zhengruifeng Originally I wrote the median function code here, but  @JoshRosen suggested extracting the common code into a util function https://github.com/apache/spark/pull/34234#discussion_r731369282 .
Either changing the median method or copying some code here all makes sense to me.
WDYT?

Reply:
ok, I am netural on it.